### PR TITLE
Bug #75576, formula can't work on freehand table (GraalJS TableArray unwrap)

### PR DESCRIPTION
**Pull request by jshobe-inetsoft**
## Summary

A viewsheet containing a formula on a freehand (calc) table hangs forever in the "loading" state when opened on the portal. This is a regression from the Rhino → GraalJS JavaScript engine migration (feature #75423).

## Root Cause

A calc-table group cell formula such as:

```js
toList(data['*@=toList(rowValue["orderdate"], "rounddate=year,interval=1.0"):$orderdate_Year'],
       'remainder=Others ,sort=desc,field=state,sorton=sum(customer_id),maxrows=3')
```

`data['*@...']` (a subtable reference — any key starting with `*`) returns a new `inetsoft.report.script.TableArray` wrapping a subtable. `FormulaFunctions.toList` calls `JavaScriptEngine.unwrap()` → `ScriptUtil.unwrap()` and relies on it converting the `TableArray` to its underlying `XTable` so the `instanceof XTable` branch (using `field=state`) runs.

Pre-migration, Rhino's `ScriptUtil.unwrap` unwrapped **any** `org.mozilla.javascript.Wrapper` via `.unwrap()`, and `TableArray` was `implements ArrayObject, Wrapper`. The GraalJS migration replaced that generic check with a special case for `inetsoft.uql.script.XTableArray` **only**, silently dropping `TableArray`.

As a result `toList` returns the raw `TableArray` as the cell value. Serializing `LoadTableDataCommand` to JSON then throws `NullPointerException` (`DFWrapper.getHeaders()` is null) in `CommandDispatcher`, so the command is never sent to the client and the viewsheet stays in the loading state.

## Fix

In `ScriptUtil.unwrap`, add a case for `inetsoft.report.script.TableArray` that unwraps it to its underlying `XTable`, mirroring the existing `XTableArray` handling. This restores the exact pre-migration Rhino unwrap semantics at the single shared unwrap chokepoint.

## Test Plan

- Import `free.vso` and open it on the portal — the freehand table now loads instead of hanging; no script-related errors in the log.
- `./mvnw clean install -pl core -am -DskipTests` → BUILD SUCCESS.
- `./mvnw test -pl core -Dtest=TableArrayTest,XTableArrayTest,FormulaFunctionsTest,FormulaFunctions2Test` → 58 tests pass.

Fixes Redmine #75576.

### CODE REVIEW

**Review by jkidacki-inetsoft — DISMISSED** [review]

File: core/src/main/java/inetsoft/util/script/ScriptUtil.java [hunk 1]
```diff
@@ -41,6 +41,16 @@ public static Object unwrap(Object obj) {
...
       }
 
+      // Likewise for a TableArray (the calc/report script table wrapper,
+      // e.g. the value returned by a calc cell's data['*@...'] subtable
+      // reference). It was also a Rhino Wrapper; unwrap it to its underlying
+      // XTable so host code (toList/mapList/etc.) can detect and process it as
+      // a table instead of leaving the non-serializable wrapper in a cell
+      // value. (#75576 / #75423)
+      if(obj instanceof inetsoft.report.script.TableArray) {
+         return ((inetsoft.report.script.TableArray) obj).unwrap();
```
[the remark this answers is not in the document]
Comment (jshobe-inetsoft) [reply core/src/main/java/inetsoft/util/script/ScriptUtil.java:51]: Thanks for the review.

1. Agreed — the `getTable()` null case only arises via the unused `TableArray(String, Class, boolean)` constructor and downstream callers already tolerate a null unwrap result, so no change needed there.
2. Added a regression test `unwrapReturnsUnderlyingXTable` in `TableArrayTest` asserting `ScriptUtil.unwrap(new TableArray(xtable))` returns the underlying `XTable` (via `assertInstanceOf`/`assertSame`), pinning this exact chokepoint.

**Comment by jshobe-inetsoft** [comment]
Thanks for the thorough review. Addressed the one actionable (non-blocking) note by adding a regression test `unwrapReturnsUnderlyingXTable` in `TableArrayTest` that pins `ScriptUtil.unwrap(TableArray)` -> underlying `XTable`. The `getTable()` null edge case is left as-is (pre-existing, only via an unused constructor, and downstream callers tolerate null).

**Review by jkidacki-inetsoft — APPROVED** [review]
